crates/wdk-build/src/cargo_make.rs

```diff
@@ -596,24 +596,36 @@ pub fn setup_path() -> Result<impl IntoIterator<Item = String>, ConfigError> {
     Ok([PATH_ENV_VAR].map(ToString::to_string))
 }
 
-// eWDK and NuGet(CI) environments set `WDKToolRoot` to the versioned directory
-// already, so `sdk_version` is added only when the env var is missing.
-fn get_wdk_tools_root(wdk_content_root: &Path, sdk_version: &str) -> Result<PathBuf, ConfigError> {
-    match env::var("WDKToolRoot") {
-        Ok(value) => Ok(PathBuf::from(value)),
-        Err(VarError::NotPresent) => Ok(wdk_content_root.join("tools").join(sdk_version)),
-        Err(e) => Err(ConfigError::EnvVarReadError("WDKToolRoot".to_string(), e)),
-    }
+fn get_wdk_tools_root(wdk_content_root: &Path, sdk_version: &str) -> PathBuf {
+    get_path_from_env("WDKToolRoot", wdk_content_root, "tools", sdk_version)
 }
 
-// eWDK and NuGet(CI) environments set `WDKBinRoot` to the versioned directory
-// already, so `sdk_version` is added only when the env var is missing.
-fn get_wdk_bin_root(wdk_content_root: &Path, sdk_version: &str) -> Result<PathBuf, ConfigError> {
-    match env::var("WDKBinRoot") {
-        Ok(value) => Ok(PathBuf::from(value)),
-        Err(VarError::NotPresent) => Ok(wdk_content_root.join("bin").join(sdk_version)),
-        Err(e) => Err(ConfigError::EnvVarReadError("WDKBinRoot".to_string(), e)),
-    }
+fn get_wdk_bin_root(wdk_content_root: &Path, sdk_version: &str) -> PathBuf {
+    get_path_from_env("WDKBinRoot", wdk_content_root, "bin", sdk_version)
+}
+
+/// Reads path from the given env variable or falls back to
+/// constructing it from WDK content root.
+///
+/// The path in `env_var` is already the full path because eWDK
+/// and Nuget CLI set it that way. In the fallback however we
+/// have to append `sub_folder` and `sdk_version` manually.
+fn get_path_from_env(
+    env_var: &str,
+    wdk_content_root: &Path,
+    sub_folder: &str,
+    sdk_version: &str,
+) -> PathBuf {
+    env::var(env_var).map_or_else(
+        |e| {
+            trace!(
```

Comment:
Not going to block merge on this, so that we can cut the patch release, but I have some thoughts here:

1. I think you should distinguish between the "env var not present" case and the "env var has invalid unicode" case
2. in the former, it should be `trace`. in the latter it should be `warn`

Reply:
Agreed